src/maze/mazeoperations.ts

```diff
@@ -0,0 +1,68 @@
+import MazePath from "./mazepath";
+import Maze from "./maze";
+import Cell from "./cell";
+
+
+// Returns an array of cells which represents the solution to the maze.
+export function getSolution(maze: Maze, mazePath: MazePath) {
+
+    let visited: boolean[][] = [];     // Keeps track of previously visited paths.
+    let toExplore: Cell[] = [];        // Location data will be stored as an array holding an x and y value.
+    let discoveredFrom: Cell[][] = []; // Marks which cell the current cell had come from.
+
+    // Clears the maze path.
+    for (let i = 0; i < maze.getColumns(); i++) {
+        visited[i] = [];
+        discoveredFrom[i] = [];
+    }
+
+    visited[0][0] = true;
+    discoveredFrom[0][0] = maze.getCell(0, 0);
+    toExplore.push(maze.getCell(0, 0));
+
+    /*
+    Runs a BFS on the maze until it reaches the endpoint.
+    As the BFS runs, it keeps track of each cell that it comes from.
+    Also keeps track of visited nodes to prevent looping.
+     */
+    while (toExplore.length) {
+        let curr = toExplore.shift();
+        let neighbors = maze.neighbours(curr)
+
+        for(let c of neighbors) {
+            let x = c.getColumn()
+            let y = c.getRow();
+
+            if (!visited[x][y] && c.isOpen()) {
+                discoveredFrom[x][y] = curr;
+                visited[x][y] = true;
+                toExplore.push(c);
+            }
+        }
+    }
+    let solutionPath: Cell[] = [];
```

Comment:
I have an idea to make the backtracking portion a bit more clean. Notice that a cell can be discovered from at most one cell. Thus, we don't need to use the array toExplore. Let's do something like:

```
let curr = maze.getCell(maze.getRows() - 1, maze.getColumns() - 1);

while (curr) {
    solutionPath.push(curr);
    curr = discoveredFrom[curr.getColumn()][curr.getRow()];
}
```